File: aes.js
'use strict';
const crypto = require('crypto');

function PKCS7(msg) {
  let b = 16 - (msg.length % 16);
  return msg + String.fromCharCode(b).repeat(b);
}

function unPKCS7(msg) {
  return msg.slice(0, -(msg.slice(msg.length-1).charCodeAt(0)));
}

function encrypt(msg, key, iv) {
  let cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
  let encryptedData = cipher.update(PKCS7(msg), 'utf8', 'hex');
  return encryptedData;
}

function decrypt(ct, key, iv) {
  let decipher = crypto.createDecipheriv('aes-128-cbc', key, iv);
  decipher.setAutoPadding(false);
  let decryptedData = decipher.update(ct, 'hex', 'utf8');
  return unPKCS7(decryptedData);
}

File: aes.py
from Crypto.Cipher import AES

def PKCS7(msg):
    b = 16 - (len(msg) % 16)
    return (msg + b * chr(b)).encode()

def unPKCS7(msg):
    return msg[:-ord(msg[len(msg)-1:])]

def encrypt(msg, key, iv):
    aes = AES.new(key.encode(), AES.MODE_CBC, iv.encode())
    return aes.encrypt(PKCS7(msg))

def decrypt(ct, key, iv):
    aes = AES.new(key.encode(), AES.MODE_CBC, iv.encode())
    return unPKCS7(aes.decrypt(ct))
